ecl/eclcc/NEWACTIVITY.rst

```diff
@@ -0,0 +1,537 @@
+  To avoid the ambiguity all trailing attributes or values need to be included in the sortList.
+* Including a transform production in the sortList elements causes problems with other transform disambiguation 
+  (e.g., DATASET[x] and AGGREGATE).
+* We could require an attribute around the transform e.g., OUTPUT(transform), but that does not really fit in with other
+  activities in the language.
+* We could change the parameter order, e.g., move the transform earlier, but that would make the syntax
+  counter-intuitive.
+* We could require { } around the list � but this is inconsistent with some of the other sort orders.
+
+In order to make some progress I elected to choose the last option and require the sort order to be included in curly 
+braces.  There are already a couple of activities � subsort and a form of atmost that similarly require them (and if 
+redesigning ECL from scratch I would be tempted to require them everywhere).  The final syntax is something that will 
+need further discussion as part of the review of the pull request though, and may need to be revisited.
+
+Having decided how to solve the ambiguities in the grammar, the following is a walkthrough of the changes that were
+made as part of commit https://github.com/ghalliday/HPCC-Platform/commit/3d623d1c6cd151a0a5608aa20ae4739a008f6e44:
+
+* no_quantile in hqlexpr.hpp
+
+  The ECL query is represented by a graph of "expression" nodes � each has a "kind" that comes from the enumeration
+  _node_operator.  The first requirement is to add a new enumeration to represent the new activity � in this case we
+  elected to reuse an unused placeholder.  (These placeholders correspond to some old operators that are no longer
+  supported, but which have not been removed because the other elements in the enumeration need to keep the same 
+  values because they are used for calculating hashes for persists and other things.)
+  
+* New attribute names in hqlatoms.
+
+  The quantile activity introduces some new attribute names that have not been used before.  All names are represented in
+  an atom table, so the code in hqlatoms.hpp/cpp is updated to define the new atoms.
+  
+* Properties of no_quantile
+
+  There are various places that need to be updated to allow the system to know about the properties of the new
+  operator:
+ 
+  - hqlattr
+ 
+    This contains code to calculate derived attributes.  The first entry in the case statement is currently unused (the
+    function should be removed).  The second inside calcRowInformation() is used to predict how many rows are generated by this
+    activity.  This information is percolated through the graph and is used for optimizations, and input counts can be
+    used to select the best implementation for a particular activity.
+   
+  - hqlexpr
+ 
+    Most changes are relatively simple, and include the text for the operator, whether it is constant, the number of
```

Comment:
Most changes are relatively simple~~,~~ and include the text for the operator, whether it is constant, **and** the number of...